pages/searching/csv_export.rst

```diff
@@ -1,17 +1,55 @@
+It is possible to export the results of your search as a CSV document. To do so, click on the three dots on the right side of the search bar and select the *Export to CSV* option.
 
 .. image:: /images/searching/csv_export.png
    :align: center
 
-This will open a modal where you can choose the fields the CSV document should contain. You can also optionally specify a stream to filter the messages.
-Clicking on *Downlaod* will create the file based on the specified options and start the download.
+This will open a modal where you can choose the fields the CSV document should contain. The initial fields are based on the message table configured for the search page.
+It is also possible to define a limit. Because messages are loaded in chunks, all chunks up to the one where the limit is reached will be retrieved.
+Which means the total number of delivered messages can be higher than the defined limit.
 
-.. image:: /images/searching/csv_export_dialog.png
+Clicking on *Start Download* will create the file based on the specified options and start the download.
+
+.. image:: /images/searching/csv_export_dialog_v2.png
+   :align: center
+
+When the search page or saved search contains multiple message tables, you can decide which message table you want to adopt the settings from.
+In this case you will see the following select when opening the CSV export modal:
+
+.. image:: /images/searching/csv_export_widget_select.png
    :align: center
 
-**Hint**: Some Graylog inputs keep the original message in the ``full_message`` field. If you need to export the original message, you
-can do so by clicking on the *List all fields* link at the bottom of the sidebar, and then selecting the ``full_message`` field.
+Additionally, you can select *Export to CSV* in a message table’s action menu, if you want to adopt its settings directly.
+
+.. image:: /images/searching/csv_export_message_table_action_menu.png
+   :align: center
+
+Exporting Message Tables on a Dashboard
+=======================================
+
+The export on a dashboard is slightly different compared to the search page or saved searches.
+Because dashboards do not have a single result set, you always need to select a message table when clicking on *Export to CSV* in a dashboard’s search bar action menu.
+You can find more information on this topic in the section :ref:`dashboards-widget-specific-criteria`.
+
+Even though a message table on dashboard has its own search criteria and the dashboard’s search bar only functions as a filter, you will always export the result set currently displayed by the message table.
+If you want to export only the search results the message table’s search criteria is referring to, make sure the dashboard’s filter is not defined.
+
+Decorator Support
+=================
+
+While the CSV export supports fields created by decorators, they are currently not being listed in the fields select options list.
```

Comment:
While the CSV export supports fields created by decorators, they are currently not being listed in the fields select options list and must be created manually.